File: Source/Aladdin.ASN1.ISO/ASN1/ISO/PKIX/X400/TeletexOrganizationalUnitNames.cs
using System;
using System.IO;

// TeletexOrganizationalUnitNames ::= SEQUENCE SIZE (1..ub-organizational-units) OF TeletexOrganizationalUnitName
// ub-organizational-units INTEGER ::= 4

// TeletexOrganizationalUnitName ::= TeletexString (SIZE (1..ub-organizational-unit-name-length))
// ub-organizational-unit-name-length INTEGER ::= 32

namespace Aladdin.ASN1.ISO.PKIX.X400
{
	public class TeletexOrganizationalUnitNames : Sequence<TeletexString>
	{
		// конструктор при раскодировании
		public TeletexOrganizationalUnitNames(IEncodable encodable) : base(encodable) 
		{
			// проверить корректность
			if (Length <= 0 || Length > 4) throw new InvalidDataException(); 

			// проверить корректность элементов
			foreach (TeletexString obj in this) TeletexString.Validate(obj, false, 1, 32); 
		} 
		// конструктор при закодировании
		public TeletexOrganizationalUnitNames(params TeletexString[] values) : base(values) 
		{
			// проверить корректность
			if (Length <= 0 || Length > 4) throw new ArgumentException(); 

			// проверить корректность элементов
			foreach (TeletexString obj in this) TeletexString.Validate(obj, true, 1, 32); 
		} 
	}
}

File: Source/Aladdin.ASN1.ISO.Java/src/aladdin/asn1/iso/pkix/x400/TeletexOrganizationalUnitNames.java
package aladdin.asn1.iso.pkix.x400;
import aladdin.asn1.*;
import java.io.*; 

// TeletexOrganizationalUnitNames ::= SEQUENCE SIZE (1..ub-organizational-units) OF TeletexOrganizationalUnitName
// ub-organizational-units INTEGER ::= 4

// TeletexOrganizationalUnitName ::= TeletexString (SIZE (1..ub-organizational-unit-name-length))
// ub-organizational-unit-name-length INTEGER ::= 32

public final class TeletexOrganizationalUnitNames extends Sequence<TeletexString>
{
	// конструктор при раскодировании
	public TeletexOrganizationalUnitNames(IEncodable encodable) throws IOException 
	{
		super(TeletexString.class, encodable); 
		
		// проверить корректность
		if (size() <= 0 || size() > 4) throw new IOException(); 

		// проверить корректность элементов
		for (TeletexString obj : this) 
		{
			// проверить корректность элемента
			TeletexString.validate(obj, false, 1, 32);
		} 
	} 
	// конструктор при закодировании
	public TeletexOrganizationalUnitNames(TeletexString... values) 
	{
		super(TeletexString.class, values); 
		
		// проверить корректность
		if (size() <= 0 || size() > 4) throw new IllegalArgumentException(); 

		// проверить корректность элементов
		for (TeletexString obj : this) 
		{
			// проверить корректность элемента
			try { TeletexString.validate(obj, true, 1, 32); }

            // пропустить невозможную ошибку
            catch (IOException e) { throw new RuntimeException(e); } 
		} 
	} 
}
